apps/web/app/(basenames)/name/[username]/opengraph-image.tsx

```diff
@@ -63,24 +68,35 @@ export default async function OpenGraphImage(props: ImageRouteProps) {
 
   const domainName = isDevelopment ? `http://localhost:3000` : 'https://www.base.org';
   const profilePicture = getBasenameImage(username);
+  const chain = getChainForBasename(username as Basename);
   let imageSource = domainName + profilePicture.src;
 
   // NOTE: Do we want to fail if the name doesn't exists?
   try {
-    const client = getBasenamePublicClient(base.id);
-    const avatar = await client.getEnsAvatar({
+    const client = getBasenamePublicClient(chain.id);
+    let avatar = await client.getEnsText({
       name: username,
-      universalResolverAddress: USERNAME_L2_RESOLVER_ADDRESSES[base.id],
-      assetGatewayUrls: {
-        ipfs: CLOUDFARE_IPFS_PROXY,
-      },
+      key: UsernameTextRecordKeys.Avatar,
+      universalResolverAddress: USERNAME_L2_RESOLVER_ADDRESSES[chain.id],
     });
 
-    // Satori Doesn't support webp
-    if (avatar && !avatar.endsWith('.webp')) {
+    if (!avatar) return;
+
+    // IPFS Resolution
+    if (IsValidIpfsUrl(avatar)) {
+      const ipfsUrl = getIpfsGatewayUrl(avatar as IpfsUrl);
+      if (ipfsUrl) {
+        imageSource = ipfsUrl;
+      }
+    } else {
       imageSource = avatar;
     }
-  } catch (error) {}
+
+    // Cloudinary resize / fetch
+    imageSource = getCloudinaryMediaUrl({ media: imageSource, format: 'png', width: 120 });
```

Comment:
Should we define 120 as a constant somewhere? I see it in more than one place.

Reply:
good catch, we actually want a different size for this one :)